accessEntity/data.go

```diff
@@ -0,0 +1,751 @@
+		case "":
+			return d.getAppByAppName(priVal, "", "", "", org)
+		}
+	case IdentifierConsumerKey:
+		return d.getAppByConsumerKey(priVal, org)
+	}
+	return
+}
+
+func (d *DbManager) GetCompanies(org, priKey, priVal, secKey, secVal string) (companies []common.Company, err error) {
+	switch priKey {
+	case IdentifierAppId:
+		return d.getCompanyByAppId(priVal, org)
+	case IdentifierCompanyName:
+		return d.getCompanyByName(priVal, org)
+	case IdentifierConsumerKey:
+		return d.getCompanyByConsumerKey(priVal, org)
+	}
+	return
+}
+
+func (d *DbManager) GetCompanyDevelopers(org, priKey, priVal, secKey, secVal string) (companyDevelopers []common.CompanyDeveloper, err error) {
+	if priKey == IdentifierCompanyName {
+		return d.getCompanyDeveloperByComName(priVal, org)
+	}
+	return
+}
+
+func (d *DbManager) GetAppCredentials(org, priKey, priVal, secKey, secVal string) (appCredentials []common.AppCredential, err error) {
+
+	switch priKey {
+	case IdentifierConsumerKey:
+		return d.getAppCredentialByConsumerKey(priVal, org)
+	case IdentifierAppId:
+		return d.getAppCredentialByAppId(priVal, org)
+	}
+	return
+}
+
+func (d *DbManager) GetDevelopers(org, priKey, priVal, secKey, secVal string) (developers []common.Developer, err error) {
+	switch priKey {
+	case IdentifierAppId:
+		return d.getDeveloperByAppId(priVal, org)
+	case IdentifierDeveloperEmail:
+		return d.getDeveloperByEmail(priVal, org)
+	case IdentifierConsumerKey:
+		return d.getDeveloperByConsumerKey(priVal, org)
+	case IdentifierDeveloperId:
+		return d.getDeveloperById(priVal, org)
+	}
+	return
+}
+
+func (d *DbManager) getApiProductsByName(apiProdName string, org string) (apiProducts []common.ApiProduct, err error) {
+	err = d.GetDb().QueryStructs(&apiProducts,
+		sql_select_api_product+
+			`WHERE ap.name = ? AND ap.tenant_id IN `+
+			sql_select_tenant_org,
+		apiProdName,
+		org,
+	)
+	return
+}
+
+func (d *DbManager) getApiProductsByAppId(appId string, org string) (apiProducts []common.ApiProduct, err error) {
+	cols := []string{"*"}
+	query := selectApiProductsById(
+		selectAppCredentialMapperByAppId(
+			"'"+appId+"'",
+			"apiprdt_id",
+		),
+		cols...,
+	) + " AND ap.tenant_id IN " + sql_select_tenant_org
+	//log.Debugf("getApiProductsByAppId: %v", query)
+	err = d.GetDb().QueryStructs(&apiProducts, query, org)
+	return
+}
+
+func (d *DbManager) getApiProductsByConsumerKey(consumerKey string, org string) (apiProducts []common.ApiProduct, err error) {
+	cols := []string{"*"}
+	query := selectApiProductsById(
+		selectAppCredentialMapperByConsumerKey(
+			"'"+consumerKey+"'",
+			"apiprdt_id",
+		),
+		cols...,
+	) + " AND ap.tenant_id IN " + sql_select_tenant_org
+	//log.Debugf("getApiProductsByConsumerKey: %v", query)
+	err = d.GetDb().QueryStructs(&apiProducts, query, org)
+	return
+}
+
+func (d *DbManager) getApiProductsByAppName(appName, devEmail, devId, comName, org string) (apiProducts []common.ApiProduct, err error) {
```

Comment:
Are you doing proper input validation on these filter values? If not there is potential SQL injection problem.

Reply:
Good catch. Let em check it.